### PR TITLE
bug #1417: Fix exclude constraint by name

### PR DESCRIPTION
**Pull request by andreimc**
check constraints couldn't be excluded.

### CODE REVIEW

File: sql/internal/sqlx/exclude.go [hunk 1]
```diff
@@ -140,6 +140,15 @@ func excludeT(t *schema.Table, pattern string) (err error) {
...
 	})
+
+	t.Attrs, err = filter(t.Attrs, func(c schema.Attr) (bool, error) {
+		t := c.(*schema.Check)
+		match, err := filepath.Match(pattern, t.Name)
+		if !match || err != nil {
+			return false, err
+		}
+		return true, nil
+	})
```
Comment (a8m) [inline sql/internal/sqlx/exclude.go:151]: Skip the attribute if it's not a `Check` type.
```suggestion
	t.Attrs, err = filter(t.Attrs, func(c schema.Attr) (bool, error) {
		c, ok := c.(*schema.Check)
		if !ok {
			return false, nil
		}
		match, err := filepath.Match(pattern, t.Name)
		if !match || err != nil {
			return false, err
		}
		return true, nil
	})
```

File: sql/internal/sqlx/exclude_test.go [hunk 2]
```diff
@@ -94,6 +94,17 @@ func TestExcludeRealm_Tables(t *testing.T) {
 	require.Empty(t, r.Schemas[2].Tables)
 }
 
+func TestExcludeRealm_Checks(t *testing.T) {
+	r := schema.NewRealm(
+		schema.New("s1").AddTables(
+			schema.NewTable("t1").AddChecks(schema.NewCheck().SetName("c1")),
+		),
+	)
+
```
Comment (a8m) [inline sql/internal/sqlx/exclude_test.go:103]:
```suggestion
```

File: cmd/atlas/go.work.sum [hunk 3]
```diff
@@ -1,149 +1,657 @@
+bazil.org/fuse v0.0.0-20200407214033-5883e5a4b512 h1:SRsZGA7aFnCZETmov57jwPrWuTmaZK6+4R4v5FUe1/c=
```
Comment (a8m) [inline cmd/atlas/go.work.sum:1]: I don't know how did you get this, but you can remove this from this PR.

**Review by a8m — COMMENTED** [review]
🚀 

Thanks for the contribution, @andreimc. I added two little comments. Please, address them, and I'll merge this.

**Comment by andreimc** [comment]
@a8m resolved :) I did a go build cmd main.go and it changed the sum, reverted that

**Comment by andreimc** [comment]
@a8m would be good to have something like a Makefile to do the linting building etc 👍  was a bit of guess work to make this change. A contributing guide would help!

**Comment by a8m** [comment]
Merged with https://github.com/ariga/atlas/pull/1426. Thanks for the contribution, @andreimc. It's appreciated 🙏

**Comment by andreimc** [comment]
Thanks mate.

**Pull request closed without merging**